theano/tensor/tests/test_sort.py

```diff
@@ -219,3 +235,247 @@ def test_argsort_grad():
+        goal = np.sort(xval)[idx]
+
+        print(np.sort(yval))
+        print(goal)
+        assert yval.dtype == goal.dtype
+        assert np.allclose(np.sort(yval), goal)
+
+    @utt.parameterized.expand(chain(
+        product(
+            (16, 61, 257),
+            (1, -1, 10, -10, 'n//2', 'n-1', '-n', '1-n'),
+            ('float32', 'int32'),
+            ('int32', 'int64')),
+        ((2049, 1337, 'float32', 'int32'),)))
+    def test_argtopk_1d(self, size, k, dtype, idx_dtype):
+        if isinstance(k, str):
+            k = eval(k.replace('n', str(size)))
+
+        x = theano.tensor.vector(name='x', dtype=dtype)
+        y = argtopk(x, k, idx_dtype=idx_dtype)
+        fn = theano.function([x], y)
+        # generate a all-unique array
+        xval = gen_unique_vector(size, dtype)
+        yval = fn(xval)
+        idx = slice(-k, None) if k > 0 else slice(-k)
+        goal = np.argsort(xval)[idx].astype(idx_dtype)
+
+        # due to uniqueness, we expect indices same
+        assert np.all(xval[np.sort(yval)] == xval[np.sort(goal)])
+
+    @utt.parameterized.expand(chain(
+        product(
+            (16, 61, 257),
+            (1, -1, 10, -10, 'n//2', 'n-1', '-n', '1-n'),
+            ('float32', 'int32'),
+            ('int32', 'int64')),
+        ((2049, 1337, 'float32', 'int32'),)))
+    def test_combined_1d(self, size, k, dtype, idx_dtype):
+        if isinstance(k, str):
+            k = eval(k.replace('n', str(size)))
+
+        x = theano.tensor.vector(name='x', dtype=dtype)
+        yv, yi = topk_and_argtopk(x, k, idx_dtype=idx_dtype)
+        fn = theano.function([x], [yv, yi])
+        # generate a all-unique array
+        xval = gen_unique_vector(size, dtype)
+        yvval, yival = fn(xval)
+        idx = slice(-k, None) if k > 0 else slice(-k)
+        goali = np.argsort(xval)[idx].astype(idx_dtype)
+        goalv = xval[goali]
+
+        # due to uniqueness, we expect indices same
+        assert np.all(xval[np.sort(yival)] == xval[np.sort(goali)])
+        assert np.allclose(np.sort(yvval), goalv)
+
+    @utt.parameterized.expand(chain(
+        product(
+            (18, 62, 258),
+            (1, -1, 'n//2'),
+            ('int32', 'float32')),
+        ((2048, 1337, 'float32'),)))
+    def test_argtopk_1d_collision(self, size, k, dtype):
+        # with non-unique kth max value
+        if isinstance(k, str):
+            k = eval(k.replace('n', str(size)))
+
+        x = theano.tensor.vector(name='x', dtype=dtype)
+        y = argtopk(x, k, idx_dtype='int32')
+        fn = theano.function([x], y)
+        xval = np.repeat(np.random.uniform(-100., 100., size=size // 2).astype(dtype), 2)
+        xval = xval[np.random.permutation(size)]
+        yval = fn(xval)
+        idx = slice(-k, None) if k > 0 else slice(-k)
+        goal = np.argsort(xval)[idx].astype('int32')
+        print(goal)
+        print(np.argsort(xval))
+        assert np.allclose(np.sort(xval[yval]), np.sort(xval[goal]))
```

Comment:
For allclose, it is better to use:

utt.asser_allclose()

This give a better error message in case of error. This does the assert and the comparison.